gnocchi/storage/ceph.py

```diff
@@ -201,13 +200,9 @@ def _pending_measures_to_process_count(self, metric_id):
         object_prefix = self.MEASURE_PREFIX + "_" + str(metric_id)
         return len(list(self._list_object_names_to_process(object_prefix)))
 
-    def list_metric_with_measures_to_process(self, size, part, full=False):
-        names = self._list_object_names_to_process()
-        if full:
-            objs_it = names
-        else:
-            objs_it = itertools.islice(names, size * part, size * (part + 1))
-        return set([name.split("_")[1] for name in objs_it])
+    def _list_metric_with_measures_to_process(self):
+        objs_it = self._list_object_names_to_process()
+        return sorted(set(name.split("_")[1] for name in objs_it))
```

Comment:
The best sorting would be done on the number of measures that a metric have actually. That'd batch the measures treatment so that'd speed things up a bit.
I'll try to change the patch to add that.

Even without any sorting this should fix the biggest issue of the current algorithm, fingers crossed.